src/Diagnostics/FxCop/Core/Design/CA1052DiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,203 @@
+            AnalyzerPowerPackRulesResources.ResourceManager,
+            typeof(AnalyzerPowerPackRulesResources));
+
+        private static readonly LocalizableString MessageFormat = new LocalizableResourceString(
+            nameof(AnalyzerPowerPackRulesResources.StaticHolderTypeIsNotStatic),
+            AnalyzerPowerPackRulesResources.ResourceManager,
+            typeof(AnalyzerPowerPackRulesResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId,
+            Title,
+            MessageFormat,
+            AnalyzerPowerPackDiagnosticCategory.Design,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
+            helpLinkUri: "http://msdn.microsoft.com/library/ms182168.aspx",
+            customTags: DiagnosticCustomTags.Microsoft);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        protected override void AnalyzeSymbol(
+            INamedTypeSymbol symbol,
+            Compilation compilation,
+            Action<Diagnostic> addDiagnostic,
+            AnalyzerOptions options,
+            CancellationToken cancellationToken)
+        {
+            if (symbol.IsStaticHolderType()
+                && !symbol.IsStatic
+                && (symbol.IsPublic() || symbol.IsProtected()))
+            {
+                addDiagnostic(symbol.CreateDiagnostic(Rule, symbol.Name));
+            }
+        }
+    }
+
+    internal static class CA1052Extensions
+    {
+        /// <summary>
+        /// Returns a value indicating whether the specified symbol is a static
+        /// holder type.
+        /// </summary>
+        /// <param name="symbol">
+        /// The symbol being examined.
+        /// </param>
+        /// <returns>
+        /// <c>true</c> if <paramref name="symbol"/> is a static holder type;
+        /// otherwise <c>false</c>.
+        /// </returns>
+        /// <remarks>
+        /// A symbol is a static holder type if it is a class with at least one
+        /// "qualifying member" (<see cref="IsQualifyingMember(ISymbol)"/>) and no
+        /// "disqualifying members" (<see cref="IsDisqualifyingMember(ISymbol)"/>).
+        /// </remarks>
+        internal static bool IsStaticHolderType(this INamedTypeSymbol symbol)
+        {
+            if (!symbol.IsReferenceType)
+            {
+                return false;
+            }
+
+            List<ISymbol> declaredMembers = symbol.GetMembers()
+                .Where(m => !m.IsImplicitlyDeclared)
+                .ToList();
+
+            List<ISymbol> qualifyingMembers = declaredMembers
+                .Where(IsQualifyingMember)
+                .ToList();
+
+            List<ISymbol> disqualifyingMembers = declaredMembers
+                .Where(IsDisqualifyingMember)
+                .ToList();
+
+            return qualifyingMembers.Count > 0 && disqualifyingMembers.Count == 0;
```

Comment:
Consider avoiding Where() and ToList() and use IEnumerable<ISymbol> and Any() instead.